lua/obsidian/commands/extract_note.lua

```diff
@@ -0,0 +1,40 @@
+local log = require "obsidian.log"
+local util = require "obsidian.util"
+
+---Extract the selected text into a new note
+---and replace the selection with a link to the new note.
+---
+---@param client obsidian.Client
+return function(client, data)
+  local viz = util.get_visual_selection()
+  if viz.lines == nil or #viz.lines == 0 then
+    log.err "ObsidianExtractLink must be called with visual selection"
+    return
+  elseif #viz.lines ~= 1 then
+    log.err "Only in-line visual selections allowed"
+    return
```

Comment:
Was this on purpose? I think it's okay to have multiple lines, right?

Reply:
You're right. I used link.lua as baseline for the code and did not take it out. Fixing it now.

Reply:
Addressed in efed66a.